libsolidity/formal/BMC.cpp

```diff
@@ -764,7 +764,8 @@ void BMC::checkCondition(
 	case smt::CheckResult::SATISFIABLE:
 	{
 		std::ostringstream message;
-		message << _description << " happens here";
+		message << _description << " happens here.";
```

Comment:
I'm not sure about this one. The idea is that if there is a counterexample the sentence goes on as `...happens here for:`, where in this case it'll be `happens here. for:`

Reply:
I assumed that's your intention but you are actually never modifying the `message` variable again in your first `if`-branch. You're using it as-is. Instead you add your example as a `SecondarySourceLocation`. Not sure if you intended it that way.

Reply:
Right, because text is just added after `message`. But in the `else` branch, where there are no counterexamples, `.` is added to `message`, that's why it doesn't have `.` before branching.

Reply:
No.. it's not. Text is _not_ added after `message`. Only `SecondarySourceLocation` "text". If the first `if` branch is taken, the `.` will be missing completely.

Reply:
Most of your tests show this, too ;) As an example, see 
https://github.com/ethereum/solidity/blob/develop/test/libsolidity/smtCheckerTests/functions/constructor_simple.sol#L16
found using `git grep "happens here$"` which results in about 360 tests.
And only two tests with the `.`